File: 백준/Silver/1991. 트리 순회/트리 순회.py
import sys

tree={}
n=int(input())
for i in range(n) :
    root, left, right = sys.stdin.readline().strip().split()
    tree[root]=[left,right]

def preorder(root) :
    if(root=='.'):
        return
    print(root,end='')
    preorder(tree[root][0])
    preorder(tree[root][1])
    return
def inorder(root) :
    if(root=='.'):
        return
    inorder(tree[root][0])
    print(root,end='')
    inorder(tree[root][1])
    return
def postorder(root) :
    if(root == '.'):
        return
    postorder(tree[root][0])
    postorder(tree[root][1])
    print(root,end='')
    return

preorder('A')
print()
inorder('A')
print()
postorder('A')

File: 백준/Silver/1991. 트리 순회/트리 순회.cc
#include<iostream>
#include<vector>
#include<map>

using namespace std;

int n;

struct Node {
	char left;
	char right;
};

map<char, Node> m;

void pre(char root) {
	if (root == '.') return;
	cout << root;
	pre(m[root].left);
	pre(m[root].right);
}
void in(char root) {
	if (root == '.') return;
	in(m[root].left);
	cout << root;
	in(m[root].right);
}
void post(char root) {
	if (root == '.') return;
	post(m[root].left);
	post(m[root].right);
	cout << root;
}

int main() {
	cin >> n;

	for (int i = 0; i < n; ++i) {
		char root, left, right;
		cin >> root >> left >> right;
		m[root].left = left;
		m[root].right = right;
	}
	
	pre('A');
	cout << "\n";
	in('A');
	cout << "\n";
	post('A');

}